src/PrestaShopBundle/Form/Admin/Configure/ShopParameters/OrderPreferences/GeneralType.php

```diff
@@ -0,0 +1,138 @@
+
+/**
+ * Class generates "General" form
+ * in "Configure > Shop Parameters > Order Settings" page.
+ */
+class GeneralType extends TranslatorAwareType
+{
+    /**
+     * @var CurrencyDataProvider
+     */
+    private $currencyDataProvider;
+
+    /**
+     * @var CMSDataProvider
+     */
+    private $cmsDataProvider;
+
+    public function __construct(
+        TranslatorInterface $translator,
+        array $locales,
+        CurrencyDataProvider $currencyDataProvider,
+        CMSDataProvider $cmsDataProvider
+    ) {
+        parent::__construct($translator, $locales);
+
+        $this->currencyDataProvider = $currencyDataProvider;
+        $this->cmsDataProvider = $cmsDataProvider;
+    }
+
+    public function buildForm(FormBuilderInterface $builder, array $options)
+    {
+        /** @var Configuration $configuration */
+        $configuration = $this->getConfiguration();
+        $isMultishippingEnabled = $configuration->getBoolean('PS_ALLOW_MULTISHIPPING');
+        $defaultLanguageId = $configuration->getInt('PS_LANG_DEFAULT');
+        $defaultCurrencyId = $configuration->getInt('PS_CURRENCY_DEFAULT');
+        $defaultCurrency = $this->currencyDataProvider->getCurrencyById($defaultCurrencyId);
+
+        $builder
+            ->add('enable_final_summary', SwitchType::class)
+            ->add('enable_guest_checkout', SwitchType::class)
+            ->add('disable_reordering_option', SwitchType::class)
+            ->add('purchase_minimum_value', MoneyWithSuffixType::class, [
+                'currency' => $defaultCurrency->iso_code,
+                'suffix' => $this->trans('(tax excl.)', 'Admin.Global'),
+            ])
+            ->add('recalculate_shipping_cost', SwitchType::class)
+        ;
+
+        if ($isMultishippingEnabled) {
+            $builder->add('allow_multishipping', SwitchType::class);
+        }
+
+        $builder
+            ->add('allow_delayed_shipping', SwitchType::class)
+            ->add('enable_tos', SwitchType::class)
+            ->add('tos_cms_id', ChoiceType::class, [
+                'choices_as_values' => true,
+                'placeholder' => $this->trans('None', 'Admin.Global'),
+                'choices'  => $this->getCMSChoices($defaultLanguageId),
+            ])
+        ;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function configureOptions(OptionsResolver $resolver)
+    {
+        $resolver->setDefaults([
+            'translation_domain' => 'Admin.Shopparameters.Feature',
+        ]);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getBlockPrefix()
+    {
+        return 'order_preferences_general_block';
+    }
+
+    /**
+     * Gets CMS choices for choice type
+     *
+     * @param $languageId
+     *
+     * @return array
+     */
+    private function getCMSChoices($languageId)
+    {
+        $choices = [];
+
+        foreach ($this->cmsDataProvider->getCMSPages($languageId) as $cms) {
```

Comment:
I'd like to see this function moved in CmsDataProvider instead, so you can inject the array instead of the full CmsDataProvider (like we already do to inject the locales).